filesystem.go

```diff
@@ -170,7 +171,12 @@ func cmdFilesystem(logger log.Logger, reg *prometheus.Registry, promClient api.C
 				case <-ctx.Done():
 					return nil
 				case f := <-files:
-					level.Debug(logger).Log("msg", "reading", "file", f)
+					// We only care about watching for files with the .yaml extension
+					if filepath.Ext(f) != ".yaml" {
+						continue
+					}
```

Comment:
I guess we can actually accept both `.yaml` and `.yml` while I prefer the four letter one and is generally the one used throughout Prometheus and Kubernetes afaik. 
```suggestion
					// We only care about watching for files with the .yaml extension
					if filepath.Ext(f) != ".yaml" && filepath.Ext(f) != ".yml" {
						level.Warn(logger).Log("msg", "ignoring non YAML file", "file", f)
						continue
					}
```